### PR TITLE
[risk=low][RW-7718] Add access module table with toggles to AdminUserProfile

### PR DESCRIPTION
**Pull request by jmthibault79**
Bypass_Toggles

---
**PR checklist**

- [x] This PR meets the Acceptance Criteria in the JIRA story
- [x] The JIRA story has been moved to Dev Review
- [ ] This PR includes appropriate unit tests
- [x] I have run and tested this change locally
- [ ] I have run the E2E tests on ths change against my local UI and/or API server with `yarn test-local` or [yarn test-local-devup](https://github.com/all-of-us/workbench/blob/main/e2e/README.md#examples)
- [ ] If this includes a UI change, I have taken screen recordings or screenshots of the new behavior and notified the PO and UX designer
- [x] If this includes an API change, I have updated the appropriate Swagger definitions and notified API consumers
- [x] If this includes a new feature flag, I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later

### CODE REVIEW

File: ui/src/app/pages/admin/admin-user-common.tsx [hunk 1]
```diff
@@ -147,19 +149,38 @@ export const getUpdatedProfileValue = (
...
 
-export const enableSave = (
+export const isBypassed = (
+  profile: Profile,
+  moduleName: AccessModule
+): boolean =>
+  !!getAccessModuleStatusByName(profile, moduleName)?.bypassEpochMillis;
+
+// would this AccessBypassRequest actually change the profile state?
+// allows un-toggling of bypass for a module
```
Comment (jmthibault79) [inline ui/src/app/pages/admin/admin-user-common.tsx:159]: does this comment make sense?  please help me make it more clear, if it does not

File: ui/src/app/pages/admin/admin-user-common.tsx [hunk 2]
```diff
@@ -147,19 +149,38 @@ export const getUpdatedProfileValue = (
...
+export const wouldUpdateBypassState = (
+  oldProfile: Profile,
+  request: AccessBypassRequest
+): boolean => isBypassed(oldProfile, request.moduleName) !== request.isBypassed;
+
+export const isChanged = (
```
Comment (jmthibault79) [inline ui/src/app/pages/admin/admin-user-common.tsx:165]: note: new version does not check for the existence of errors.  this allows it to be used for more purposes besides "is the save button enabled"

File: ui/src/app/pages/admin/admin-user-profile.tsx [hunk 3]
```diff
@@ -105,8 +119,8 @@ const UneditableFields = (props: { profile: Profile }) => {
         <div style={styles.uneditableFieldsSpacer} />
         <UneditableField label='User name' value={username} />
         <UneditableField
-          label='Access Tiers'
-          value={accessTierShortNames.map(displayNameForTier).join(' ,')}
+          label='Data Access Tiers'
```
Comment (jmthibault79) [inline ui/src/app/pages/admin/admin-user-profile.tsx:122]: these 2 lines are unrelated minor fixes

File: ui/src/app/pages/admin/admin-user-profile.tsx [hunk 4]
```diff
@@ -411,7 +540,14 @@ export const AdminUserProfile = (spinnerProps: WithSpinnerOverlayProps) => {
           </ErrorsTooltip>
           <Button
             type='secondary'
-            onClick={() => setUpdatedProfile(oldProfile)}
+            disabled={
+              !isChanged(oldProfile, updatedProfile, bypassChangeRequests)
+            }
+            onClick={() => {
+              setBypassChangeRequests([]);
```
Comment (jmthibault79) [inline ui/src/app/pages/admin/admin-user-profile.tsx:547]: better behavior

**Review by aweng98 — APPROVED** [review]
New toggle looks great. I don't find anything wrong.
One nit: If the whole row highlights when the mouse is over the toggle, it would improve the look and feel.

File: ui/src/app/pages/admin/admin-user-common.tsx (hunk 2 above)
Comment (jmthibault79) [reply ui/src/app/pages/admin/admin-user-common.tsx:165]: could you say more about that?

Comment (aweng98) [reply ui/src/app/pages/admin/admin-user-common.tsx:165]: Something little more verbose along the lines of `hasBypassChanged` or `isBypassChanged` to make it more obvious.

Comment (jmthibault79) [reply ui/src/app/pages/admin/admin-user-common.tsx:165]: ok - it's the profile which has changed here.  So it can "profileNeedsUpdate()" or something
